### PR TITLE
Show info about the minikube linux distribution

### PR DESCRIPTION
**Pull request by afbjorklund**
Taking from /etc/os-release, via libmachine provision

For #4523

### CODE REVIEW

**Comment by medyagh** [comment]
there is a lint issue 
```
cmd/minikube/cmd/start.go:546:53: unnecessary conversion (unconvert)
	osReleaseInfo, err := provision.NewOsRelease([]byte(osReleaseOut))
	                                                   ^
cmd/minikube/cmd/start.go:269:16: Error return value of `showOsRelease` is not checked (errcheck)
		showOsRelease()
		             ^
```

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
I like this PR.  I do agree that showing the distro is of most importance for the none drivers, but I would also like to show it for other platforms. We have the perfect space for it here:

`😄  minikube v1.1.1 on linux (amd64)`

What I propose is to place it here:

`😄  minikube v1.1.1 on CentOS 4.5 (amd64)`

In the future I would also like to extend this to Mac OS X & Windows as well, but surprisingly, I can't find any good libraries for doing so.

**Comment by tstromberg** [comment]
@minikube-bot OK to test

This can be merged once lint issues are addressed.

**Comment by afbjorklund** [comment]
> I like this PR. I do agree that showing the distro is of most importance for the none drivers, but I would also like to show it for other platforms. We have the perfect space for it here:
> 
> `😄 minikube v1.1.1 on linux (amd64)`
> 
> What I propose is to place it here:
> 
> `😄 minikube v1.1.1 on CentOS 4.5 (amd64)`

They are talking about different things. One is showing information about the `minikube` binary (including the version and GOOS/GOARCH, which is defined to be one of "linux"/"windows/"darwin" and "amd64")

The other is showing information from the libmachine provisioner, i.e. what is being installed on the VM. Currently I used the same information also for the `none` driver, maybe that is what lead to the confusion.

**Comment by afbjorklund** [comment]
> In the future I would also like to extend this to Mac OS X & Windows as well, but surprisingly, I can't find any good libraries for doing so. That said, I'm perfectly OK with this approach as a stop-gap measure.

Most other operating systems have this in their `uname`, since they don't have as many vendors as GNU/Linux does. So you would use the go equivalent of uname(2), with the "sysname" and "release"

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
Curious failure:

```
 16:30:48 | > * Creating none VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
 16:33:49 | ! E0701 16:33:49.321193   12847 start.go:633] StartHost: DetectProvisioner: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
 16:33:49 | ! I0701 16:33:49.321250   12847 utils.go:129] non-retriable error: DetectProvisioner: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
 16:33:49 | ! W0701 16:33:49.321383   12847 exit.go:100] Unable to start VM: DetectProvisioner: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
 16:33:49 | ! X Unable to start VM: DetectProvisioner: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
 16:33:49 | ! * Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
```

Also, the Travis timeout makes me wonder if a unit test is starting up a real minikube rather than using mocks.

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
Looks like this one has a merge conflict now :(

**Comment by afbjorklund** [comment]
Ok, rebase time perhaps

**Comment by afbjorklund** [comment]
Moved into a proper function

**Comment by tstromberg** [comment]
go tests are still timing out in travis, so I wonder if additional mocking may be required?

```
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/kubeadm	0.049s	coverage: 33.1% of statements
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

**Comment by afbjorklund** [comment]
> go tests are still timing out in travis, so I wonder if additional mocking may be required?

Yes, there is something wrong with DetectProvisioner. Think we have seen this elsewhere too ?

`Waiting for SSH to be available...`

**Comment by afbjorklund** [comment]
Maybe should scope out the provisioned linux distribution for now. There's only one, anyway...

**Comment by afbjorklund** [comment]
Here is where it was seen before: #4132

```
⌛  Waiting for SSH access ...
Waiting for SSH to be available...
Getting to WaitForSSH function...
Error getting ssh command 'exit 0' : driver does not support ssh commands
```

**Comment by afbjorklund** [comment]
It seems that these tests have _always_ been timing out, since they use VirtualBox and not Mock host...

Before:
`ok  	k8s.io/minikube/pkg/minikube/cluster	360.409s	coverage: 50.6% of statements`

After:
`ok  	k8s.io/minikube/pkg/minikube/cluster	0.467s	coverage: 47.7% of statements`

That is, we seem to be calling SSH a lot for drivers (`none`, `unknown`) that don't actually support it ?

**Comment by tstromberg** [comment]
> /retest was checking translations, and some weird sshutil error ?

I think we've now addressed cluster_test, sshutil, and translation check issues. Sorry about the instability there!

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -162,9 +162,10 @@ func configureHost(h *host.Host, e *engine.Options) error {
 		if err := h.ConfigureAuth(); err != nil {
 			return &util.RetriableError{Err: errors.Wrap(err, "Error configuring auth on host")}
 		}
-		return ensureSyncedGuestClock(h)
+		if h.Driver.DriverName() != "mock" {
```
Comment (tstromberg) [inline pkg/minikube/cluster/cluster.go]: I believe the changes to cluster.go can now be reverted.

**Comment by afbjorklund** [comment]
Seems there is a bug in the MockDriver, where it returns `nil` information with `nil` error. Thanks.

**Pull request merged**